src/cli/cmds/synth.ts

```diff
@@ -53,26 +105,98 @@ class Command implements yargs.CommandModule {
         }
       });
     } else {
-      const manifests = await synthApp(command, outdir, stdout, recordConstructMetadata);
+      let manifests: SynthesizedApp;
+
+      if (format === SynthesisFormat.HELM) {
+        await createHelmScaffolding(chartApiVersion, chartVersion, outdir);
+        const templateDir = path.join(outdir, 'templates');
+
+        manifests = await synthApp(command, templateDir, stdout, recordConstructMetadata);
+      } else {
+        manifests = await synthApp(command, outdir, stdout, recordConstructMetadata);
+      }
+
       if (validations) {
         const pluginManager = new PluginManager(pluginsDir);
         await validateApp(manifests, stdout, validations, pluginManager, reportFile);
       }
     }
-
-
   }
 
 }
 
 async function fetchValidations(): Promise<ValidationConfig[] | undefined> {
-  if (typeof(config.validations) === 'string') {
+  if (typeof(config?.validations) === 'string') {
     const content = await download(config.validations);
     return yaml.parse(content) as ValidationConfig[];
   } else {
-    return config.validations;
+    return config?.validations;
   }
 }
 
+async function createHelmScaffolding(apiVersion: string, chartVersion: string, outdir: string) {
+  const tempHelmStructure = createFolderStructure();
+
+  const substituteValues = {
+    apiVersion: apiVersion,
+    version: chartVersion,
+    app: path.basename(path.resolve()),
+  };
+
+  try {
+    await sscaff(tempHelmStructure, outdir, substituteValues);
+  } finally {
+    fs.rmSync(tempHelmStructure, { recursive: true });
+  }
+
+
+  if (apiVersion === HelmChartApiVersion.V2 && crdsArePresent(config?.imports)) {
+    await addCrdsToHelmChart(outdir);
+  }
+
+  function createFolderStructure(): string {
+    const root = fs.mkdtempSync(path.join(os.tmpdir(), 'helm-scaffolding-'));
+
+    fs.mkdirSync(path.join(root, 'templates'));
+
+    const chartYamlFile = {
+      apiVersion: '{{ apiVersion }}',
+      name: '{{ app }}',
+      version: '{{ version }}',
+      description: 'Generated chart for {{ app }}',
+      type: 'application',
+    };
+
+    fs.outputFileSync(path.join(root, CHART_YAML_FILE), yaml.stringify(chartYamlFile));
+
+    const readmeFile = 'This Helm chart is generated using cdk8s. Any manual changes to the chart would be discarded once cdk8s app is synthesized again with `--format helm`.';
+
+    fs.outputFileSync(path.join(root, README), readmeFile);
+
+    if (config?.synth?.chartApiVersion === HelmChartApiVersion.V2 && crdsArePresent(config.imports)) {
```

Comment:
Notice that this *should* have been a bug because you don't consider the default value of `chartApiVersion` here.

```suggestion
    if (apiVersion === HelmChartApiVersion.V2 && crdsArePresent(config?.imports)) {
```

The only reason things work is because you use `fs.outputFileSync` when actually adding the CRDs, which is recursively creating the necessary directories. So we can get rid of this entirely.

Reply:
Hmm, wow, sorry I missed this. Curious to learn, how did you catch this? :)

Reply:
What caught my eye is the use of `config` in a function that accepts inputs that are already derived from the config. In general, there should be a single place in the code that handles every layer, here you had two.